fjpgtt/pollitos-league-manager
# Francisco league

Francisco League

Changes made:

- refashioning of variables and methods to the English language
- A specific format is applied to all classes 
- removal of unnecessary comments
- log implementation
- elimination of non-essential methods
- populated profile correction

How to test:
Once the code is running, it must go to the route http://localhost:8080 in postman, once there place the following routes to verify the functionality, it is important to take into account that for correct functioning the following creation order is followed 
1 player creation
2 team creation
3 match creation

* **Create team** = http://localhost:8080/Team/
{
"idteam": 1,
  "idLeague": 2,
  "teamname": "Equipo A"
}
{
"idteam": 2,
  "idLeague": 2,
  "teamname": "Equipo B"
}
note: it is important to create 2 teams to play a match
*** Create player indicating the team it is part of** = http://localhost:8080/Player/
{
  "idpLayer": 2,
  "idleague": 2,
  "idteam": 2,
  "nameplayer": "hermenejildo Pérez"
}
*** Create a match**= http://localhost:8080/Match/
{
  "idmatch": 1,
  "idleague": 2,
  "localteam": {
    "idteam": 1
  },
  "visitteam": {
    "idteam": 2
  },
  "goallocal": 2,
  "goalVisit": 2
}
*** Get all the teams**=http://localhost:8080/Team/
*** Get all the players from a team** = http://localhost:8080/Player/{idteam}
*** Get all the matches from a team**= http://localhost:8080/Match/team/{idteam} 
*** Edit the player information** = http://localhost:8080/Player/{idplayer}
   {
        "idJugador": 1,
        "idEquipo": 1,
        "idLiga": 1,
        "nombreJugador": "pablo Pérez"
    }
*** Edit the team information** = http://localhost:8080/Team/{idteam}
{
"idteam": 1,
  "idLeague": 2,
  "teamname": "Equipo A"
}
*** Delete all the matches** = http://localhost:8080/Match/matches
*** Delete all the players of a team** = http://localhost:8080/Team/{idteam}

Once the data has been filled out or using the populated profile, the league information can be viewed in the console
![imagen](https://github.com/user-attachments/assets/3f2a3b19-e4e9-4044-8fae-3112fedbe2eb)
